code/modules/admin/sql_message_system.dm

```diff
@@ -219,6 +232,48 @@
 			browse_messages(target_ckey = target_ckey, agegate = TRUE)
 	qdel(query_find_edit_expiry_message)
 
+/proc/edit_message_severity(message_id, browse)
+	if(!SSdbcore.Connect())
+		to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+		return
+	message_id = text2num(message_id)
+	if(!message_id)
+		return	
+	var/datum/DBQuery/query_find_edit_note_severity = SSdbcore.NewQuery("SELECT type, targetckey, adminckey, severity FROM [format_table_name("messages")] WHERE id = [message_id] AND deleted = 0")
+	if(!query_find_edit_note_severity.warn_execute())
+		qdel(query_find_edit_note_severity)
+		return
+	if(query_find_edit_note_severity.NextRow())
+		var/list/severity_list = list("High Severity" = 4, "Medium Severity" = 3, "Minor Severity" = 2, "No Severity" = 1)
+		var/type = query_find_edit_note_severity.item[1]
+		var/target_ckey = query_find_edit_note_severity.item[2]
+		var/admin_ckey = query_find_edit_note_severity.item[3]
+		var/old_severity = query_find_edit_note_severity.item[4]
+		if(old_severity)
+			var/list/num2severity = list("4" = "high", "3" = "medium", "2" = "minor", "1" = "none") // there probably is a better way to do this - CitrusGender
+			old_severity = num2severity["[old_severity]"]
+		else
+			old_severity = "NA"
+		var/editor_ckey = sanitizeSQL(usr.ckey)
+		var/new_severity = input("Set the severity of the note.", "Severity", null, null) as null|anything in severity_list
+		if(!new_severity)
+			return
```

Comment:
Need to add a qdel for `query_find_edit_note_severity` here otherwise it won't get deleted if this return happens.